lib/PGcore.pm

```diff
@@ -538,17 +542,27 @@ sub record_unlabeled_array_name {
 }
 
 sub store_persistent_data {    # will store strings only (so far)
-	my $self    = shift;
-	my $label   = shift;
-	my @content = @_;
+	my ($self, $label, @values) = @_;
 	if (defined($self->{PERSISTENCE_HASH}->{$label})) {
 		warn "can' overwrite $label in persistent data";
 	} else {
-		$self->{PERSISTENCE_HASH}->{$label} = join("", @content);    #need base64 encoding?
+		$self->{PERSISTENCE_HASH_UPDATED}->{$label} = 1;
+		$self->{PERSISTENCE_HASH}->{$label}         = join("", @values);
```

Comment:
There is one thing that I mentioned before that seems to have been forgotten.  That is the extra dereferencing arrows.  There is no need for the second level of arrows.  Change this to
```suggestion
		$self->{PERSISTENCE_HASH_UPDATED}{$label} = 1;
		$self->{PERSISTENCE_HASH}{$label}         = join("", @values);
```

Reply:
Sorry - fixed now.

Reply:
I will get this change in https://github.com/openwebwork/pg/pull/506 tomorrow.